src/main/rx/xbus.c

```diff
@@ -80,17 +87,89 @@
 // Use formula: 800 + value * 1400 / 4096 (i.e. a shift by 12)
 #define XBUS_CONVERT_TO_USEC(V) (800 + ((V * 1400) >> 12))
 
+// XBus Mode A additions
+#define XBUS_MODEA_CHANNEL_COUNT 16
+// Frame is: Frame(1 byte) + Length(1 byte) + Key and type (2 bytes - not used) + 16*channel(4 bytes) + CRC(1 byte)
+#define XBUS_MODEA_FRAME_SIZE 69
+#define XBUS_MODEA_OFFSET_BYTES 4
+#define XBUS_MODEA_BAUDRATE 250000
+#define XBUS_MODEA_MAX_FRAME_TIME 2700 // 2760us round up to 2800 (69/[250k/10bits]=2760)
+#define XBUS_MODEA_START_OF_FRAME_BYTE (0xA4)
+// Mode A needs a different conversion as the resolution is 16bit, not 12bit
+//0x0000      800uSec
+//0x1249      900uSec(-60°, -75°, or -90°)
+//0x7FFF      1500uSec(0°)
+//0xEDB6      2100uSec(+60°, +75°, or +90°)
+//0xFFFF      2200uSec
+#define XBUS_MODEA_CONVERT_TO_USEC(V) (800 + ((V * 1400) >> 16))
+// As Mode A has a difference frame time, lets set it in a global
+static uint32_t xBusMaxFrameTime;
+
 static bool xBusFrameReceived = false;
 static bool xBusDataIncoming = false;
 static uint8_t xBusFramePosition;
 static uint8_t xBusFrameLength;
 static uint8_t xBusChannelCount;
 static uint8_t xBusProvider;
 
-
 // Use max values for ram areas
-static volatile uint8_t xBusFrame[XBUS_FRAME_SIZE_A2];  //size 35 for 16 channels in xbus_Mode_B
-static uint16_t xBusChannelData[XBUS_RJ01_CHANNEL_COUNT];
+static volatile uint8_t xBusFrame[XBUS_MODEA_FRAME_SIZE];  // size 69 for 16 channels in xbus_Mode_A
+static uint16_t xBusChannelData[XBUS_MODEA_CHANNEL_COUNT];
+
+// xBus Mode A uses an 8-bit CRC based on Dallas/Maxim
```

Comment:
Please don't use large data tables, they flash space is precious.
Instead, use the CRC functions in common/crc.h instead.